lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -434,6 +438,41 @@ APInt getNextInputVal(Inst *Var,
   llvm::report_fatal_error("Model does not contain the guess input variable");
 }
 
+Inst *findConst(souper::Inst *I, std::string ConstName,
+                std::set<const Inst *> &Visited) {
+  if (I->Name == ConstName) {
+    return I;
+  } else {
+    for (auto &&Op : I->Ops) {
+      if (Visited.find(Op) == Visited.end()) {
+        auto Ret = findConst(Op, ConstName, Visited);
```

Comment:
```
if (findConst(Op, ConstName, Visited))
  return true;
```

Reply:
It needs to return the constant it found recursively, can't shorten to something like that.

Reply:
I'm saying get rid of the useless temporary called "Ret"

Reply:
Ret has 2 uses, in the if condition and then in the return. Can't get rid of it.

Reply:
the code you have looks like this:

```
auto Ret = findConst(Op, ConstName, Visited);
if (Ret) {
  return Ret;
}
```

I am asking you to remove the useless temporary, and rewrite the code like this:
```
if (findConst(Op, ConstName, Visited)) {
  return true;
}
```
I can't think of any way to say this more clearly.

Reply:
The return type is an Inst *, not a bool here.

Reply:
ah, carry on then